### PR TITLE
NavBar Links shown appropriately based off logged in status and active based off page loaded.

### LINKED ISSUE #49: Improper re-direct from Documentation to Dashboard when not signed in
**Scenario**
I am not logged in and I click on the Documentation tab and it takes me to the documentation page. When I click on the 'Dashboard' link (that shouldn't be there if I am not signed in) it takes me to the Developer dashboard and then it re-directs me to the sign in page
image

**Expected Result**
I should be taken directly to the Sign in Page

Actual Result
I am taken to the Developer Dashboard for a second and then taken to the sign in page. See video below
http://www.screencast.com/t/t3hLbkcVE

### LINKED ISSUE #50: Navigation Bar Links when Not Signed In
**Scenario**
I am on the Documentation page and I am not signed in

**Expected Result**
- I should not see the Dashboard link on the navigation bar. I should see: 
   To the left - Harvest API | Documentation | ABIS Home Page 
   To the right - Sign In | Sign Up 

- The Harvest API link should take me to the Harvest home page https://harvest-api-staging.herokuapp.com/#/

**Actual Result**

-  I see a "Documentation" link on the navigation bar
image

- When  I click on the Harvest API link, nothing happens

### LINKED ISSUE #56: Dashboard highlighted when on the documentation page
Scenario
I am signed in on the developer dashboard and i click on the Documentation tab

Expected Result
I am taken to the the Documentation page and I see the Documentation tab highlighted

Actual Result
The Dashboard tab is highlighted
image

### LINKED ISSUE #89: Creation of "Apps" Dashboard Link for the Administration
Scenario
========
When an admin user is logged in there is no way to access the "admin dashboad" from the navigation panel. Clicking the "dashboard" menu item takes the user to the Apps Dashboard

Solution
========
* Create a new menu item called "My Apps" and point it to the Apps dashboard
* Point the "Dashboard" menu item to the Admin dashboard
* Ensure that the "Dashboard" menu item is only visible for a user that is an admin user (also ensure that the link is secured)

### LINKED ISSUE #96: "Dashboard" Navbar Links for Anonymous users
Scenario
=======
Currently any user that comes to the HarvestAPI site or Documentation page will see a link to the "Dashboard" in the navigation bar. However, the "Dashboard" is only accessible to a user who is logged in.

Expected Result
=============
The "Dashboard" like is only shown to users that are logged into HarvestAPI

### PR DESCRIPTION
**Pull request by leechristophermurray**
The list of features this branch introduces are as follows.
* based off the page loaded, the appropriate navbar link will be highlighted
    * _to facilitate this, the **docs is now loaded as an iframe**._
    * changed the height to 88% in the style.css to remove the double scrollbars on the documentation page.
* `Harvest API` link now returns to the homepage
* ~~`Dashboard`~~ `My Apps` link is now shown only when logged in. This was implemented by adding `ng-if="userLoggedIn"` to the link tags attributes. *The developer dashboard is now labeled as **"My Apps"** as requested on #89.*
*  `Dashboard` link is now a link to the admin dashboard and shown **only** when and admin user is logged in. This was implemented by adding `ng-if="current_user.us_user_role == 'admin'"` to the link tags attributes.

This: 
* fixes #96
* fixes #50 
* fixes #56 
* fixes #49 
* fixes #89

**THIS BRANCH FIXES ALL THE ISSUES RELATED TO THE NAVBAR**

### CODE REVIEW

File: views/docs.ejs [hunk 1]
```diff
@@ -30,12 +30,12 @@
```
Comment (matjames007) [inline views/docs.ejs:30]: @thinkking is this file being used? -if not remove it

File: public/javascript/controllers/user-ctrl.js [hunk 2]
```diff
@@ -202,6 +226,10 @@ angular.module('harvestv2')
                 });
             };
 
+            $scope.isActive = function (linkPath) {
```
Comment (matjames007) [inline public/javascript/controllers/user-ctrl.js:211]: @thinkking What does this do?  Comment the code.

File: public/javascript/controllers/user-ctrl.js [hunk 3]
```diff
@@ -86,6 +86,30 @@ angular.module('harvestv2')
 
         }
     ]
+).controller("DocsCtrl", ['$scope', '$location', '$routeParams', 'UserFactory', 'RolesFactory',
+        'PlatformFactory', 'CurrentUserFactory',
+        function($scope, $location, $routeParams, UserFactory, RolesFactory, PlatformFactory, CurrentUserFactory) {
+
```
Comment (matjames007) [inline public/javascript/controllers/user-ctrl.js]: @thinkking writing jQuery within angular framework is bad.  Attempt to rewrite this controller using Angular functions.

File: models/db.js [hunk 4]
```diff
@@ -6,7 +6,7 @@
 var mongoose = require('mongoose');
 var Schema = mongoose.Schema;
 
-mongoose.connect(process.env.MONGOLAB_URI);
+mongoose.connect(process.env.MONGOLAB_URI || "localhost:27017/mongodb");
```
Comment (matjames007) [inline models/db.js]: @thinkking change this.

**Review by matjames007 — CHANGES_REQUESTED** [review]
Comments made within the in-line commenting feature.  Please review.

File: views/docs.ejs (hunk 1 above)
Comment (leechristophermurray) [reply views/docs.ejs:30]: Yes, it is. The original `/docs` path to the documentation is still available but is now being used to load the docs as an iframe as described in the pull request (#133) description. This had to be done to facilitate the docs' tab being shown as active as angular will not detect a path outside the app.

**Comment by leechristophermurray** [comment]
@matjames007 this pull request may be reviewed again.

**Comment by matjames007** [comment]
@thinkking merge in `develop`.

Issues:

- iFrame is a bad way for presenting the documentation and presents additional issues when the page is scaled to a tablet display
- Instead of making one call to load the page, two calls are made to the server with twice the calls for all the page resources used by the application.  This means that an instance of the angular application is running with an instance of the docs.ejs page inside the iframe with almost all the same scripts, styling files and html elements.
- Please ensure that you cover all the changes you did in the commit messages.

This being said - i suggest you demonstrate the functionality to another core team member for feedback on whether this can be merge in as part of the MVP.

**Comment by matjames007** [comment]
Not sure what happened with those commits @thinkking .... but I think the .yaml file is broken and doesn't render well in redoc.

**Comment by leechristophermurray** [comment]
@matjames007 and @nick22891 this pull request may be reviewed again. The docs are showing as they should due to a few lines of styling I had to do to the `main.css` we have for our docs.

File: models/db.js [hunk 5]
```diff
@@ -6,7 +6,7 @@
 var mongoose = require('mongoose');
 var Schema = mongoose.Schema;
 
-mongoose.connect(process.env.MONGOLAB_URI);
+mongoose.connect(process.env.MONGOLAB_URI ||  "localhost:27017/mongodb");
```
Comment (matjames007) [inline models/db.js]: @thinkking change again....

**Comment by leechristophermurray** [comment]
@matjames007 the mongoose.connect() arguments have been reverted.

**Review by matjames007 — APPROVED** [review]

**Pull request merged**